src/overlays/actors/ovl_Bg_Hidan_Firewall/z_bg_hidan_firewall.h

```diff
@@ -4,12 +4,29 @@
 #include <ultra64.h>
 #include <global.h>
 
-typedef struct
+struct BgHidanFirewall;
+
+typedef void FirewallFunc(struct BgHidanFirewall* this, GlobalContext* globalCtx);
+
+typedef struct BgHidanFirewall
 {
-    /* 0x0000 */ Actor actor;
-    /* 0x014C */ char unk_14C[0x54];
+    /* 0x0000 */ Actor                 actor;
+    /* 0x014C */ FirewallFunc*         actionFunc;
```

Comment:
as of now we use ActorFunc for action functions. This is something we may start doing eventually, but we want to all make that change at the same time to keep things consistent